loopd/swapclient_server.go

```diff
@@ -885,9 +885,12 @@ func (s *swapClientServer) SuggestSwaps(ctx context.Context,
 			return nil, err
 		}
 
+		clonedPubkey := route.Vertex{}
+		copy(clonedPubkey[:], pubkey[:])
```

Comment:
Great catch!

Reply:
I too would have thought that `pubkey[:]` clones the bytes. I'm still confused why not 😄

Reply:
It's a shallow copy, basically just a pointer to the same array. see example here: https://go.dev/play/p/1G4bvUPulE_l